plugins/modules/ndo_port_channel_interface.py

```diff
@@ -0,0 +1,403 @@
+            if int(range_stop) > int(range_start):
+                for x in range(int(range_start), int(range_stop) + 1):
+                    interface_ids.append("{0}/{1}".format(slots, x))
+            else:
+                errors_interfaces.append(interface)
+        elif re.fullmatch(r"((\d+/)+\d+$)", interface):
+            interface_ids.append(interface)
+        else:
+            errors_interfaces.append(interface)
+    return set(interface_ids), errors_interfaces
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True),
+        port_channel_interface=dict(type="str", aliases=["name", "port_channel"]),
+        port_channel_interface_uuid=dict(type="str", aliases=["uuid", "port_channel_uuid"]),
+        description=dict(type="str"),
+        node=dict(type="str"),
+        interfaces=dict(type="list", elements="str", aliases=["members"]),
+        interface_policy=dict(type="str", aliases=["policy", "pc_policy"]),
+        interface_policy_uuid=dict(type="str", aliases=["policy_uuid", "pc_policy_uuid"]),
+        interface_descriptions=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                interface_id=dict(type="str"),
+                description=dict(type="str"),
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["template", "port_channel_interface"]],
+            ["state", "present", ["template", "port_channel_interface"]],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    port_channel_interface = module.params.get("port_channel_interface")
+    port_channel_interface_uuid = module.params.get("port_channel_interface_uuid")
+    description = module.params.get("description")
+    node = module.params.get("node")
+    interfaces = ",".join(module.params.get("interfaces")) if module.params.get("interfaces") else None
+    interface_policy = module.params.get("interface_policy")
+    interface_policy_uuid = module.params.get("interface_policy_uuid")
+    interface_descriptions = module.params.get("interface_descriptions")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "fabric_resource", template)
+    mso_template.validate_template("fabricResource")
+    object_description = "Port Channel Interface"
+
+    path = "/fabricResourceTemplate/template/portChannels"
+    existing_template = mso_template.template.get("fabricResourceTemplate", {}).get("template", {})
+    existing_port_channel_interfaces = existing_template.get("portChannels") if existing_template.get("portChannels") else []
```

Comment:
existing_template.get("portChannels") if existing_template.get("portChannels") else [] ->

existing_template.get("portChannels", [])

Reply:
`existing_template.get("portChannels", [])` would be wrong cause the `get(key, alt_value)` method will only return `alt_value` if the `key` does not exist. But if the `key` does exist but the value attached to it is None, it will return None whatever `alt_value` is. That's why `existing_template.get("portChannels") if existing_template.get("portChannels") else []` is safer in case the key is None or the value attached to it is None as we want `existing_port_channel_interfaces` to be of type list

Reply:
still existing_template.get("portChannels") doesn't have to be called twice.